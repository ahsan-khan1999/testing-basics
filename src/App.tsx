import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
  
        <a>
          Learn React
        </a>
        <input placeholder='press me'  style={{height:80,width:150}}/>
      
    </div>
  );
}

export default App;
