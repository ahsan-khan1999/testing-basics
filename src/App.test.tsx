import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
test('another test',() => {
  const {getAllByPlaceholderText} = render(<App />)
  const element = getAllByPlaceholderText(/press me/i);
  expect(element).toBeInTheDocument();
})