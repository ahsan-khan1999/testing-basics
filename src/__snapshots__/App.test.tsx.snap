// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="App"
>
  <a>
    Learn React
  </a>
  <input
    placeholder="press me"
    style={
      Object {
        "height": 80,
        "width": 150,
      }
    }
  />
</div>
`;
